woocommerce/woocommerce
# Fix subscription grouping

### Changes proposed in this Pull Request:

This PR is part of the My subscriptions redesign feature.

When a subscription is inactive, but installed, it will now show up in the installed section.
If there are multiple available subscriptions, only the first available subscription will be shown in the installed section.
All available subscriptions will have the connect button.

If there are multiple subscriptions the active subscription will be shown in the installed section.

The install button won't show for subscriptions that are already installed.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Connect your test site to Woo.com test customer account
2. [Manually install Facebook for WooCommerce](http://localhost:8888/wp-admin/plugin-install.php?s=facebook&tab=search&type=term)
3. [Open the subscriptions list](http://localhost:8888/wp-admin/admin.php?page=wc-admin&tab=my-subscriptions&path=%2Fextensions)
4. Confirm that one license is shown in the installed list and the other in not in use
5. All licenses should have a connect button
6. Connect one license
7. One license should be listed in the installed list, other licenses should stay in the not in use section and have no action buttons (install, connect...)
8. [Manually install Bookings](https://woo.com/my-account/downloads/)
9. It should be listed in the Installed section with a subscribe button
10. Try coming up with other combinations of installed and connected and confirm that they behave as expected



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

Fix subscription grouping to ensure one installed subscription is available in the installed section.

</details>
![Screenshot 2023-11-08 at 13-37-38 Extensions ‹ WooCommerce ‹ woocommerce](https://github.com/woocommerce/woocommerce/assets/1199991/6894bd0c-3945-4004-ba20-efb8468aa03e)
